File: MainApp/src/main/java/com/sleticalboy/learning/accounts/sync/RoutineSyncService.java
package com.sleticalboy.learning.accounts.sync;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import androidx.annotation.Nullable;

public final class RoutineSyncService extends Service {

    private static final Object LOCK = new Object();
    private static RoutineSyncAdapter sSyncAdapter;

    @Override
    public void onCreate() {
        super.onCreate();
        synchronized (LOCK) {
            if (sSyncAdapter == null) {
                sSyncAdapter = new RoutineSyncAdapter(getApplicationContext());
            }
        }
    }

    @Nullable
    @Override
    public IBinder onBind(final Intent intent) {
        return sSyncAdapter.getSyncAdapterBinder();
    }
}

File: MainApp/src/main/java/com/sleticalboy/learning/accounts/sync/RoutineSyncService.kt
package com.sleticalboy.learning.accounts.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder

class RoutineSyncService : Service() {

    override fun onCreate() {
        super.onCreate()
        synchronized(LOCK) {
            if (sSyncAdapter == null) {
                sSyncAdapter = RoutineSyncAdapter(applicationContext)
            }
        }
    }

    override fun onBind(intent: Intent): IBinder? {
        return sSyncAdapter!!.syncAdapterBinder
    }

    companion object {
        private val LOCK = Any()
        private var sSyncAdapter: RoutineSyncAdapter? = null
    }
}